src/components/cursor.js

```diff
@@ -33,60 +38,118 @@ module.exports.Component = registerComponent('cursor', {
   dependencies: ['raycaster'],
 
   schema: {
+    downEvents: {default: []},
     fuse: {default: utils.device.isMobile()},
-    fuseTimeout: {default: 1500, min: 0}
+    fuseTimeout: {default: 1500, min: 0},
+    laserAlwaysShow: {default: true},
+    laserColor: {default: '#74BEC1'},
+    laserEnabled: {default: false},
+    laserWidth: {default: 0.001},
+    upEvents: {default: []}
   },
 
   init: function () {
-    var el = this.el;
-    var canvas = el.sceneEl.canvas;
     this.fuseTimeout = undefined;
     this.mouseDownEl = null;
     this.intersection = null;
     this.intersectedEl = null;
 
-    // Wait for canvas to load.
-    if (!canvas) {
-      el.sceneEl.addEventListener('render-target-loaded', bind(this.init, this));
-      return;
-    }
-
     // Bind methods.
-    this.onMouseDown = bind(this.onMouseDown, this);
-    this.onMouseUp = bind(this.onMouseUp, this);
+    this.onCursorDown = bind(this.onCursorDown, this);
+    this.onCursorUp = bind(this.onCursorUp, this);
     this.onIntersection = bind(this.onIntersection, this);
     this.onIntersectionCleared = bind(this.onIntersectionCleared, this);
+  },
 
-    // Attach event listeners.
-    canvas.addEventListener('mousedown', this.onMouseDown);
-    canvas.addEventListener('mouseup', this.onMouseUp);
-    el.addEventListener('raycaster-intersection', this.onIntersection);
-    el.addEventListener('raycaster-intersection-cleared', this.onIntersectionCleared);
+  update: function (oldData) {
+    var data = this.data;
+    var el = this.el;
+    var raycaster = el.components.raycaster.raycaster;
+    var rayLength;
+
+    // Create laser beam.
+    if (data.laserEnabled) {
+      rayLength = raycaster.far === Infinity ? 1000 : raycaster.far;
+      el.setAttribute('line', {
+        color: data.laserColor,
+        start: raycaster.ray.origin,
+        end: endVec3.copy(raycaster.ray.origin).add(
+          directionVec3.copy(raycaster.ray.direction).multiplyScalar(rayLength)
+        ),
+        width: data.laserWidth
+      });
+    }
+
+    if (data.laserEnabled && oldData.laserEnabled) { el.removeAttribute('line'); }
```

Comment:
forgive me if I'm wrong, but won't this incorrectly remove the [`line` just set above](https://github.com/aframevr/aframe/pull/2678/files#diff-e9782f61decda1df23d89fcb932be260R73) (if `laserEnabled` is truthy on this component `update`)?